Sources/QuickTests/Helpers/XCTestObservationCenter+QCKSuspendObservation.m

```diff
@@ -0,0 +1,23 @@
+@import XCTest;
+#import <objc/runtime.h>
+
+@interface XCTestObservationCenter (Redeclaration)
+- (NSMutableSet *)observers;
+@end
+
+@implementation XCTestObservationCenter (QCKSuspendObservation)
+
+static BOOL (^isFromApple)(id, BOOL *) = ^BOOL(id observer, BOOL *stop){
+    return [[NSBundle bundleForClass:[observer class]].bundleIdentifier containsString:@"com.apple.dt.XCTest"];
+};
+
+- (void)qck_suspendObservationForBlock:(void (^)(void))block {
+    NSSet *observersToSuspend = [[self observers] objectsPassingTest:isFromApple];
+    [[self observers] minusSet:observersToSuspend];
+
+    block();
+
+    [[self observers] unionSet:observersToSuspend];
```

Comment:
Just a thought: it might be technically possible that if `block` raises an exception this line won't run, leaving the `observers` set in an unexpected state. Probably a real edge case, even if it's possible.

Just wondering whether it would be worth wrapping this in a try/finally, and restoring the observers before rethrowing the exception?

Sorry for the drive-by comment!